spring-boot-project/spring-boot-test/src/main/java/org/springframework/boot/test/web/SpringBootTestRandomPortEnvironmentPostProcessor.java

```diff
@@ -57,7 +57,7 @@ public void postProcessEnvironment(ConfigurableEnvironment environment,
 		}
 		Integer serverPort = getPropertyAsInteger(environment, SERVER_PORT_PROPERTY,
 				8080);
-		if (!managementPort.equals(serverPort)) {
+		if (!managementPort.equals(serverPort) || managementPort == 0) {
```

Comment:
@nosan I think we'd be able to achieve this by adding a `|| managementPort.equals(0)` [here](https://github.com/spring-projects/spring-boot/blob/f5eb853f263ad96ee304749525cd7a853c8c18f2/spring-boot-project/spring-boot-test/src/main/java/org/springframework/boot/test/web/SpringBootTestRandomPortEnvironmentPostProcessor.java#L55)

Reply:
@mbhave thank you